### PR TITLE
Prevent using Yoda-style conditionals

### PR DESCRIPTION
**Pull request by pamil**
It's hard to see the benefit for Yoda-style conditionals in a codebase using `===` instead of `=`. Instead, it makes code look like `if (0 < $value)`, which is read like `if zero is less than the value`.

### CODE REVIEW

**Comment by lchrusciel** [comment]
I used to them, and the argument about not making mistakes also convince me. The third argument in favour of it is [Symfony recommendation](https://symfony.com/doc/current/contributing/code/standards.html#structure). On the other hand, we have tests for this kind of problems, and also it is probably less readable for newcomers.

To sum up, I'm against this fixer, but I will not fight for that. If you decide to go for it, I would prefer to add PHPStan to our testing pipeline and block variable assignment in `if` conditions. 

PS. Sorry for misclick.

File: tests/Sample.php [hunk 1]
```diff
@@ -4,9 +4,14 @@
 
 namespace Tests\CodingStandard;
 
-class Sample
+final class Sample
```
Comment (lchrusciel) [inline tests/Sample.php:7]: Adding final keyword should be a part of separate PR :)

**Comment by Zales0123** [comment]
I'm not sure about that. On the one hand, I think it's good to force such practices as Yoda conditions, as they really help less experienced developers to make fewer mistakes. On the other hand, I agree that it looks totally unnatural 😄 I would vote for 👍 if we also prevent setting variables in `if` statements (double protection 🛡)

**Comment by pamil** [comment]
Yeah, agreed with your opinions. I'm thinking about enforcing Yoda-style conditions for `==` and `===` and enforcing non Yoda-style conditions for operators like `>` / `<` etc., wdyt? Anyway, closing for now :)
